spec/unit/provider/osx_profile_spec.rb

```diff
@@ -0,0 +1,251 @@
+#
+# Author:: Nate Walck (<nate.walck@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Chef::Provider::OsxProfile do
+  let(:shell_out_success) do
+    double('shell_out', :exitstatus => 0, :error? => false)
+  end
+  describe 'action_create' do
+    before(:each) do
+      @node = Chef::Node.new
+      @events = Chef::EventDispatch::Dispatcher.new
+      @run_context = Chef::RunContext.new(@node, {}, @events)
+
+      @new_resource = Chef::Resource::OsxProfile.new("Profile Test", @run_context)
+      @provider = Chef::Provider::OsxProfile.new(@new_resource, @run_context)
+      @all_profiles = {"_computerlevel"=>
+          [{"ProfileDisplayName"=>"ScreenSaver Settings",
+            "ProfileIdentifier"=>"com.apple.screensaver",
+            "ProfileInstallDate"=>"2015-10-05 23:15:21 +0000",
+            "ProfileItems"=>
+             [{"PayloadContent"=>
+                {"PayloadContentManagedPreferences"=>
+                  {"com.apple.screensaver"=>
+                    {"Forced"=>[{"mcx_preference_settings"=>{"idleTime"=>0}}]}}},
+               "PayloadDisplayName"=>"Custom: (com.apple.screensaver)",
+               "PayloadIdentifier"=>"com.apple.screensaver",
+               "PayloadType"=>"com.apple.ManagedClient.preferences",
+               "PayloadUUID"=>"73fc30e0-1e57-0131-c32d-000c2944c108",
+               "PayloadVersion"=>1}],
+            "ProfileOrganization"=>"Facebook",
+            "ProfileRemovalDisallowed"=>"false",
+            "ProfileType"=>"Configuration",
+            "ProfileUUID"=>"1781fbec-3325-565f-9022-8aa28135c3cc",
+            "ProfileVerificationState"=>"unsigned",
+            "ProfileVersion"=>1},
+          {"ProfileDisplayName"=>"ScreenSaver Settings",
```

Comment:
The identifier is different, which is what matters. I'll update the DisplayName so it is also different (having the same displayname is very confusing).